### ISSUE TITLE
How to plot random(group-based scores) effects?

### ISSUE THREAD

**Issue by fatihbozdag** [arviz-devs/arviz#2188]
## Short Description

Greetings all,

I'm looking for a way to plot differences based on random effects on categorical regression. However, I failed to find any examples or API reference.

For instance;

## Code Example or link

```
import arviz as az
import bambi as bmb
import matplotlib.pyplot as plt
import numpy as np
import pandas as pd
import seaborn as sns
import pymc as pm
import pymc.sampling_jax
import numpyro

df = pd.DataFrame({"Target": target_variable, "Random_Effects_Var": random_effects, "Column1": column1, "Column2": column2})

model = bmb.Model("Target~ Column1 + Column2 + (1|Random_Effects_Var)",
                  data = df, family = "categorical", auto_scale = True, priors = priors)

model.build()

with model.backend.model:
    idata=pm.sampling_jax.sample_numpyro_nuts(draws= 250, tune = 50, target_accept = .99,
                                              postprocessing_backend = 'cpu', idata_kwargs={"log_likelihood":True},chain_method='parallel')
    posterior_predictive = pm.sample_posterior_predictive(trace = idata, extend_inferencedata=True)

idata
```
So far, so good. Imagine, Random_Effect_Var includes 2 levels of categorical data, A and B , and I'd like to plot the influence of each (Indeed group-level scores) over the Target variable along with variables in Column1 and Column2. 

here is an example from R package `Modelbased `[link](https://github.com/easystats/modelbased)

<img width="857" alt="Screenshot 2022-12-28 at 02 03 15" src="https://user-images.githubusercontent.com/59308858/209734989-39830b7a-3c70-4ad4-9afe-f0f7d4b40c03.png">

How can I plot something similar with `Arviz`?

**Comment by OriolAbril** [comment]
Could you share the InferenceData you get? On [figshare](https://figshare.com/) for example? This way we'll get a better idea and will be able to make some tests wihout needing to generate all the data ourselves and trying to get it to look like yours.

I am not sure there is a clear way to do this now. But I think it will be possible, and it will also be very helpful to ensure it  easy to do in future versions.

**Comment by fatihbozdag** [comment]
Sure thing. I've pickled the data, you can find it [here](https://doi.org/10.6084/m9.figshare.21789347)

**Comment by fatihbozdag** [comment]
meanwhile, model is as follows; 

```
Formula: Modal~ 0 + Subject_Pos + Pattern_Type + Modal_C + Verb_C + Title + (1|Native_language)
Family name: Categorical
Link: softmax
Observations: 14330
Priors:
  Common-level effects
    Subject_Pos ~ StudentT(mu: 0, nu: 30, lam: 1)
    Pattern_Type ~ StudentT(mu: 0, nu: 30, lam: 1)
    Modal_C ~ StudentT(mu: 0, nu: 30, lam: 1)
    Verb_C ~ StudentT(mu: 0, nu: 30, lam: 1)
    Title ~ StudentT(mu: 0, nu: 30, lam: 1)
  Group-level effects
    1|Native_language ~ Normal(mu: 0, sigma: HalfNormal(sigma: 2.5))
```

```
idata.posterior_predictive
Out[6]: 
<xarray.Dataset>
Dimensions:      (chain: 4, draw: 250, Modal_dim_2: 14330)
Coordinates:
  * chain        (chain) int64 0 1 2 3
  * draw         (draw) int64 0 1 2 3 4 5 6 7 ... 243 244 245 246 247 248 249
  * Modal_dim_2  (Modal_dim_2) int64 0 1 2 3 4 ... 14325 14326 14327 14328 14329
Data variables:
    Modal        (chain, draw, Modal_dim_2) int64 1 6 0 1 3 2 0 ... 7 7 8 8 7 8
Attributes:
    created_at:                 2022-12-29T10:40:55.988242
    arviz_version:              0.14.0
    inference_library:          pymc
    inference_library_version:  5.0.1
```

and

```
idata.posterior_predictive
Out[6]: 
<xarray.Dataset>
Dimensions:      (chain: 4, draw: 250, Modal_dim_2: 14330)
Coordinates:
  * chain        (chain) int64 0 1 2 3
  * draw         (draw) int64 0 1 2 3 4 5 6 7 ... 243 244 245 246 247 248 249
  * Modal_dim_2  (Modal_dim_2) int64 0 1 2 3 4 ... 14325 14326 14327 14328 14329
Data variables:
    Modal        (chain, draw, Modal_dim_2) int64 1 6 0 1 3 2 0 ... 7 7 8 8 7 8
Attributes:
    created_at:                 2022-12-29T10:40:55.988242
    arviz_version:              0.14.0
    inference_library:          pymc
    inference_library_version:  5.0.1
idata.posterior
Out[7]: 
<xarray.Dataset>
Dimensions:                      (chain: 4, draw: 250, Subject_Pos_dim: 2,
                                  Modal_dim: 8, Pattern_Type_dim: 4,
                                  Modal_C_dim: 2, Verb_C_dim: 8, Title_dim: 50,
                                  Native_language__factor_dim: 2)
Coordinates:
  * chain                        (chain) int64 0 1 2 3
  * draw                         (draw) int64 0 1 2 3 4 ... 245 246 247 248 249
  * Subject_Pos_dim              (Subject_Pos_dim) <U4 'NOUN' 'PRON'
  * Modal_dim                    (Modal_dim) <U6 'could' 'may' ... 'would'
  * Pattern_Type_dim             (Pattern_Type_dim) <U23 'Modal + be + Ving' ...
  * Modal_C_dim                  (Modal_C_dim) <U11 'per/pos/abi' 'vol/pre'
  * Verb_C_dim                   (Verb_C_dim) <U25 'Aspectual' ... 'Occurrence'
  * Title_dim                    (Title_dim) <U212 'air pollution' ... 'viole...
  * Native_language__factor_dim  (Native_language__factor_dim) <U7 'Chinese' ...
Data variables:
    Subject_Pos                  (chain, draw, Subject_Pos_dim, Modal_dim) float64 ...
    Pattern_Type                 (chain, draw, Pattern_Type_dim, Modal_dim) float64 ...
    Modal_C                      (chain, draw, Modal_C_dim, Modal_dim) float64 ...
    Verb_C                       (chain, draw, Verb_C_dim, Modal_dim) float64 ...
    Title                        (chain, draw, Title_dim, Modal_dim) float64 ...
    1|Native_language_offset     (chain, draw, Native_language__factor_dim, Modal_dim) float64 ...
    1|Native_language_sigma      (chain, draw) float64 3.318 2.227 ... 3.463
    1|Native_language            (chain, draw, Native_language__factor_dim, Modal_dim) float64 ...
Attributes:
    created_at:     2022-12-29T10:40:20.054657
    arviz_version:  0.14.0
```

**Comment by fatihbozdag** [comment]
is there any update on the topic? or should I simply prepare two different models and plot them separately?

**Comment by fatihbozdag** [comment]
So far, I've tried many examples and failed miserably. 

I've applied violin plots to compare the same variable from two inference data. It would be great if I could add values for a reference level for the response. 

![Figure 2023-02-04 012447](https://user-images.githubusercontent.com/59308858/216723471-39e374ea-5d9a-46cc-95dc-7b31a8985c55.png)

I'm still looking for a way to plot posterior predictive (probabilities) per random variable.
